### PR TITLE
adds genetic powers (theres a new one since you last looked)

### PR DESCRIPTION
**Pull request by tralezab**
## About The Pull Request

tongue spike: removes tongue to shoot it outwards, embedding for a decent amount of damage.
stimmed: generic mutation, doesn't do anything
tongue spike + stimmed = chem spike: the tongue spike no longer deals a lot of damage and does not hurt to keep in. the person who fired it can activate it remotely to transfer all of his chemicals into the spike'd person

lay web: lay a "genetic" web that only lets you pass through without resistance. neato

## Why It's Good For The Game

crew content, genetics getting a bit stale and now it supports a huge pool of chems that aren't filled. on top of that, embedding is a terribly unused mechanic and mutation combos are also terribly unused

## Changelog
:cl: armhulen
add: new genetic powers are in the pool: tongue spike, chem spike, stimmed, and web laying
/:cl:

### CODE REVIEW

**Comment by knacker48** [comment]
So say if I shoot my tongue at someone, does it grow back or do I need surgery to put a new one in?

**Comment by tralezab** [comment]
surgery

**Comment by Ratvar** [comment]
So, stealthy 68+embed memes ranged brute projectile with the downside of geneticist/assistant who wasnt talking anyway going mute? Hmm.

**Comment by itseasytosee** [comment]
Sounds cool shoot your tongue, chug some cyanide, profit.

**Comment by tralezab** [comment]
@Ratvar i didn't do the math on purpose because that comment is just total bullshit. this is a weaker shuriken from my testing, and shurikens deal ~30 damage. beats me why the hell pain multiplier ADDS.

**Comment by Ohagi-Chan** [comment]
Could think of it this way:

You only get one shot.
Having a missing tongue is a dead giveaway that you may have used this - you can't just pretend you're mute because of a trauma etc.

One concern is that adding 2 base traits, 1 of which only exists to introduce 1 more combo... this is a bit iffy.  It bugs me already that some of the existing combos use mutually exclusive bases.  Combo systems feel better when the base ingredients overlap more.

Is tongue spike a hidden gene belonging to moths?

**Comment by wesoda25** [comment]
chem spike seems kinda unbalanced but it'll probably be so rarely used it won't matter. Definitely look forward to geneticist traitor with that

**Comment by knacker48** [comment]
While I'm not a big fan of powers that do nothing on their own like strength, If it means we get more powers for genetics I'll happily take it.

File: code/datums/mutations/actions.dm [hunk 1]
```diff
@@ -230,3 +230,140 @@
...
+	var/obj/item/hardened_spike/spike = new spike_path(get_turf(C), C)
+	tongue.forceMove(spike)
+	spike.throw_at(get_edge_target_turf(C,C.dir), 14, 4, C)
+
+/obj/item/hardened_spike
+	name = "biomass spike"
+	desc = "Hardened biomass, shaped into a spike. Very pointy!"
+	icon_state = "tonguespike"
+	force = 2
+	throwforce = 15 //15 + 2 (WEIGHT_CLASS_SMALL) * 4 (EMBEDDED_IMPACT_PAIN_MULTIPLIER) = i didnt do the math
```
Comment (ATH1909) [inline code/datums/mutations/actions.dm]: that's hardly a "fairly large chunk of damage"...

File: code/datums/mutations/body.dm [hunk 2]
```diff
@@ -227,6 +227,13 @@
 	text_gain_indication = "<span class='notice'>You feel strong.</span>"
 	difficulty = 16
 
+/datum/mutation/human/stimmed
+	name = "Stimmed"
+	desc = "The user's chemical balance is more robust."
+	quality = POSITIVE
+	text_gain_indication = "<span class='notice'>You feel stimmed.</span>"
```
Comment (ATH1909) [inline code/datums/mutations/body.dm]: Maybe add some passive self-chem purging or something to this mutation or something? It'd feel kind of lame to be working on a mutation for a while, only to roll... this.

**Comment by ATH1909** [comment]
I kind of feel like these mutations are going to be completely unused, especially due to their relatively high instability costs. Reduce the instability cost of these down to, like, 5 (for the straight up brute damage one) or 10 (for the chem one), and _maybe_ these'll see some use, but no one will be happy to roll them (due to how hard it is to "reload" them and because losing a tongue is a horrific downside that no one will want to endure unless they're sure that they're about to die).

Can these be used while cuffed?

What if you give people who have either of the two tongue-firing mutations the ability to (re)insert into their mouths tongues that they click on themselves with? It'd sort of be like skeletons with their limb-reattachment trait, but (just) for tongues instead of for limbs. Plus, the imagery of a geneticist harvesting enough tongues from corpses to form an "ammo box" of tongues is pretty cool.

**Comment by MimicFaux** [comment]
Adds an interesting ranged alternative for changlings. spending chems to reload tongues.

File: code/datums/mutations/actions.dm [hunk 3]
```diff
@@ -230,3 +230,140 @@
...
+
+/datum/action/innate/send_chems/Activate()
+	if(!ishuman(transfered) || !ishuman(owner))
+		return
+	var/mob/living/carbon/human/transferer = owner
+
+	to_chat(transfered, "<span class='warning'>You feel a tiny prick!</span>")
+	transferer.reagents.trans_to(transfered, transferer.reagents.total_volume, 1, 1, 0, transfered_by = transferer)
+
+	var/obj/item/bodypaspikert/L = spikey.loc
```
Comment (Jordie0608) [inline code/datums/mutations/actions.dm]: I'm not sure what this type path is meant to be but it isn't this.

**Comment by Iatots** [comment]
Not really sure about the chemspike stuff. You only get one shot, which means you have to make it count, but it transfers chems remotely on command? So basically, you can shoot your tongue out at someone, run away immediately and either go get a new tongue because you failed, or down 3 or 4 pills and immediately transfer your contents to your opponent if you hit them.

**Comment by Ohagi-Chan** [comment]
Something to consider with chemspike is the interaction between them and "aheal" slime cores, which will grow your tongue back.  It's a bit of a waste, but highly useful for stinging multiple targets back to back.

File: code/datums/mutations/actions.dm (hunk 1 above)
Comment (tralezab) [reply code/datums/mutations/actions.dm]: pain damage on shurikens are crazy just take my word on this, this 15 damage procs over and over and over

File: code/datums/mutations/body.dm (hunk 2 above)
Comment (tralezab) [reply code/datums/mutations/body.dm]: i'll add some more combinations but this is pretty much exactly like strength

File: code/datums/mutations/actions.dm (hunk 3 above)
Comment (tralezab) [reply code/datums/mutations/actions.dm]: AAAAH

**Comment by tralezab** [comment]
see people are coming up with ways to work around the organ loss. i'm very happy with you guys that i didn't really need to add a spoonfed way to keep using it.

**Comment by tralezab** [comment]
new power added because maintainers took too long and i repeatedly conflict, it lets you lay webs like a spider

**Comment by bobbahbrown** [comment]
this is an awesome pr great job

**Comment by Tlaltecuhtli** [comment]
> Something to consider with chemspike is the interaction between them and "aheal" slime cores, which will grow your tongue back. It's a bit of a waste, but highly useful for stinging multiple targets back to back.

maybe the problem is just the availability of aheals 👁 👃 👁

**Comment by necromanceranne** [comment]
Actually wouldn't mind if this was an actual changeling power since spitting a spike at someone is super thematic for them. But antag freeze, alas.

**Comment by Ohagi-Chan** [comment]
Speaking of, it would be pretty funny to sting a ling with their own chems.
Sure, they can just purge them, but think of the funny side.

**Comment by necromanceranne** [comment]
Oooh, saucy. This is also a very handy means of quickly purging your chems too, so that's a very solid point. Pulling a fast one like that sounds hilariously funny.

**Comment by itseasytosee** [comment]
laying webs seems a bit less cool but it might have some neat combos.

File: code/game/objects/effects/spiders.dm [hunk 4]
```diff
@@ -37,16 +38,37 @@
 
 /obj/structure/spider/stickyweb/CanAllowThrough(atom/movable/mover, turf/target)
 	. = ..()
-	if(istype(mover, /mob/living/simple_animal/hostile/poison/giant_spider))
+	if(!genetic)
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/objects/effects/spiders.dm]:
```DM
if(genetic)
    return
```

Comment (ExcessiveUseOfCobblestone) [reply code/game/objects/effects/spiders.dm]: you don't even need this var though, you've already subtyped it?

File: code/game/objects/effects/spiders.dm [hunk 5]
```diff
@@ -37,16 +38,37 @@
...
-		if(istype(mover.pulledby, /mob/living/simple_animal/hostile/poison/giant_spider))
+	else if(isliving(mover)) //we change the spider to not be able to go through here
+		if(mover.pulledby == allowed_mob)
 			return TRUE
 		if(prob(50))
 			to_chat(mover, "<span class='danger'>You get stuck in \the [src] for a moment.</span>")
 			return FALSE
-	else if(istype(mover, /obj/projectile))
-		return prob(30)
+	//we don't have projectile here because that's the only case that would get this far, meaning the above layer already did it. am i making this weird, explaining . = ..()?
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/objects/effects/spiders.dm]: projectiles aren't handled by ..() if genetics is true

File: code/game/objects/effects/spiders.dm (hunk 4 above)
Comment (tralezab) [reply code/game/objects/effects/spiders.dm]: definitely more readable

File: code/game/objects/effects/spiders.dm (hunk 5 above)
Comment (tralezab) [reply code/game/objects/effects/spiders.dm]: true doe true doe i'll just add it again

**Comment by Ty-the-Smonk** [comment]
Make stimmed interact with more genes

**Comment by tralezab** [comment]
in time i will Ty

File: code/datums/mutations/actions.dm [hunk 6]
```diff
@@ -230,3 +230,189 @@
...
+	name = "chem spike"
+	desc = "Hardened biomass, shaped into... something."
+	icon_state = "tonguespikechem"
+	throwforce = 2 //2 + 2 (WEIGHT_CLASS_SMALL) * 0 (EMBEDDED_IMPACT_PAIN_MULTIPLIER) = i didnt do the math again but very low or smthin
+	embedding = list("embedded_pain_multiplier" = 0, "embed_chance" = 100, "embedded_fall_chance" = 0, "embedded_pain_chance" = 0, "embedded_ignore_throwspeed_threshold" = TRUE) //never hurts once it's in you
+	var/been_places = FALSE
+	var/datum/action/innate/send_chems/chems
+
+/obj/item/hardened_spike/chem/embedded(mob/living/carbon/human/embedded_mob)
+	if(!been_places)
```
Comment (ExcessiveUseOfCobblestone) [inline code/datums/mutations/actions.dm]:
```DM
	if(been_places)
	    return
```

**Review by ExcessiveUseOfCobblestone — COMMENTED** [review]
ping when fixed

**Comment by PKPenguin321** [comment]
Could use a slightly more verbose changelog

**Pull request merged**